### PR TITLE
Week 6 alex ballou

### PR DESCRIPTION
**Pull request by necessities**
# Submitting Week 6 Assessment

### ✅ Completed Files
Select the following as confirmation that the files in this repository are complete.
- [✅ ] Coding Challenges
- [✅ ] Coding Concepts
- [ ✅] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -24,6 +30,58 @@ const people = [
...
+
+  }
+  // console.log(result)
+  // console.log(result[0])
+  return [result[0], result[1], result[2]]
+}
+
+describeObject(people)
+
+// There was alot of trial and error for this one; initially i made the jest test to test for three seperate strings but i had to redo that and make it test for the one array output; also it took a while to remember to use dot notation on the objects to retreive their values. finally i had to edit the string interpolation the function returned to keep adding string pieces i had forgotten to make the test passed
```
Comment (elysemontano) [inline code-challenges.test.js:83]: Great work, this definitely is a tricky one that takes some trial and error.

File: code-challenges.test.js [hunk 2]
```diff
@@ -38,6 +96,57 @@ const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
...
+  console.log(returnNumbers2)
+  let finalArray = returnNumbers2.map(value => value % 3)
+  console.log(finalArray)
+  return finalArray
+  
+}
+
+arrayOfRemainders(hodgepodge2)
+
+// both tests are now passing, I had to change how the numbers were filtered a few times to make the first test pass; then check to see if the number was a real number with the .isFinite method to make the second pass
```
Comment (elysemontano) [inline code-challenges.test.js:147]: Interesting approach!  There is also a built-in method called typeof that can be utilized inside the filter to evaluate if the typeof is a number.

File: code-challenges.test.js [hunk 3]
```diff
@@ -49,3 +158,31 @@ const cubeAndSum2 = [0, 5, 10]
...
+
+describe("returnCubedArray", () => {
+  it("takes in an array of numbers and returns the sum of all the numbers cubed", () => {
+    expect(returnCubedArray(cubeAndSum1)).toEqual(99),
+    expect(returnCubedArray(cubeAndSum2)).toEqual(1125)
+  })
+})
+
+// test failed with function not defined
+// I need to make a function called returnCubedArray that takes in one array; cubes all items in the array then adds them together and returns that number
```
Comment (elysemontano) [inline code-challenges.test.js:173]: Make sure to expand on your pseudocode, meaning talk about the different methods you are using to get there.  If you use iteration, you should specify that.

File: interview-questions.md [hunk 4]
```diff
@@ -8,36 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: You will need to generate a new migration file with an add_column line, then do a new db:migrate command to create the column and add it to the table. The command could be: rails generate migration add_cohort_to_students cohort_id:integer 
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: The Post and Destroy and routes must always be passed params, these two specifically need params to make a new object and also to specify which object to delete. 
 
-Researched answer:
+Researched answer: According to the guide on rails website, Patch, Put and Delete will always need an ID passed into the path on the route.
```
Comment (elysemontano) [inline interview-questions.md:19]: The show route will also get a param as we need to know which individual instance to show.

File: interview-questions.md [hunk 5]
```diff
@@ -8,36 +8,48 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+1. As a user I want to be a ble to view a list of things i have to do.
+2. As a user i want to be able to add to the list of things i have to do.
+3. As a user i want to be able to mark an item complete.
+4. As a user i want to be able to edit an item. 
+5. As a user i want to be able to add sub items to an item.
+6. As a user i want to be able to add a time to each item.
+7. As a user i want to be able to send my list to other people to see.
+8. As a user i want to be able to customize how the list looks.
+9. As a user i want to be able to have some items re occur automatically
+10. As a user i want to be able to have a calendar view of my items.
```
Comment (elysemontano) [inline interview-questions.md:54]: Excellent work on this section!

File: rails-commenting.rb [hunk 6]
```diff
@@ -54,15 +54,15 @@ def destroy
...
       redirect_to blog_post_path(@post)
     end
   end
 
-  # ---9)
+  # ---9) this private keyword keeps everything after local to the application
   private
   def blog_post_params
-    # ---10)
+    # ---10) this require method is saying that it is required that the object type is a blog_post and to only permit the two data type of title and content to be passed into it
```
Comment (elysemontano) [inline rails-commenting.rb:65]: Awesome work!!

**Review by elysemontano — APPROVED** [review]
Great work Alex!  Your interview answers and rails commenting section is solid, and your code logic looks great.  Just make sure to expand on your pseudocode, especially when in interviews.  Awesome job!

File: code-challenges.test.js (hunk 1 above)
Comment (necessities) [reply code-challenges.test.js:83]: Thank you!

**Pull request merged**